widgets/deployments/src/widget.tsx

```diff
@@ -1,16 +1,48 @@
-// @ts-nocheck File not migrated fully to TS
-
 import { get, isEmpty, isEqual } from 'lodash';
+import type {
+    PageSizeConfiguration,
+    PollingTimeConfiguration,
+    SortAscendingConfiguration,
+    SortColumnConfiguration
+} from 'app/utils/GenericConfig';
+import './widget.css';
+import type { PaginatedResponse } from 'backend/types';
+import type { DeploymentViewData } from 'widgets/deployments/src/DeploymentViewTypes';
+import type { Execution } from 'app/utils/shared/ExecutionUtils';
+import type { InstanceSummaryItem } from 'app/widgets/common/nodes/NodeInstancesConsts';
+import type { Deployment } from 'app/widgets/common/deploymentsView/types';
 import DeploymentsList from './DeploymentsList';
 import FirstUserJourneyButtons from './FirstUserJourneyButtons';
-import './widget.css';
-
-const t = Stage.Utils.getT('widgets.deployments');
-
-Stage.defineWidget({
+import type { DeploymentViewDataWithSelected } from './DeploymentViewTypes';
+
+const translate = Stage.Utils.getT('widgets.deployments');
+
+export interface DeploymentsConfiguration
+    extends PollingTimeConfiguration,
+        PageSizeConfiguration,
+        SortColumnConfiguration,
+        SortAscendingConfiguration {
+    blueprintIdFilter: string;
+    showFirstUserJourneyButtons: boolean;
+    clickToDrillDown: boolean;
+    displayStyle: 'table' | 'list';
+    showExecutionStatusLabel: boolean;
+}
+
+type DeploymentParams = {
```

Comment:
To be inline with other types let's have:
```suggestion
type DeploymentsParams = {
```